pkg/resourcemanager/pollclient/pollclient.go

```diff
@@ -125,3 +127,53 @@ func (client PollClient) GetResponder(resp *http.Response) (result PollResponse,
 	result.Response = autorest.Response{Response: resp}
 	return
 }
+
+type LongRunningOperationPollResult string
+
+const (
+	PollResultNoPollingNeeded       = LongRunningOperationPollResult("noPollingNeeded")
+	PollResultCompletedSuccessfully = LongRunningOperationPollResult("completedSuccessfully")
+	PollResultTryAgainLater         = LongRunningOperationPollResult("tryAgainLater")
+)
+
+func (client PollClient) PollLongRunningOperationIfNeeded(ctx context.Context, status *v1beta1.ASOStatus) (LongRunningOperationPollResult, error) {
+	// Before we attempt to issue a new update, check if there is a previously ongoing update
+	if status.PollingURL == "" {
+		return PollResultNoPollingNeeded, nil
+	}
+
+	res, err := client.Get(ctx, status.PollingURL)
+	pollErr := errhelp.NewAzureError(err)
+	if pollErr != nil {
+		if pollErr.Type == errhelp.OperationIdNotFound {
+			// Something happened to our OperationId, just clear things out and try again
+			status.PollingURL = ""
+		}
+		return PollResultTryAgainLater, err
+	}
+
+	if res.Status == LongRunningOperationPollStatusFailed {
+		// TODO: Do we really want to be setting message in here?
+		status.Message = res.Error.Error()
```

Comment:
I think this is a good idea, even if it's only there for a short time - it gives the user a chance to notice something when there's something going wrong.

Reply:
The TODO here was mostly me getting at the awkwardness of setting properties on the passed in type, but we're already doing it for PollingURL too and I'm not sure that there's a cleaner way to do this.

I'm removing this TODO because I agree with you that for now this is the best thing to do.